### PR TITLE
Copy text to PRIMARY, not CLIPBOARD in visual mode

### PR DESCRIPTION
**Pull request by hpdeifel**
Previously, the selected text in visual mode was copied to the X
CLIPBOARD, while 'normal' Emacs copies the active region to the
PRIMARY selection (controlled by `select-active-regions`). The latter
is also what the major X toolkits do by default.

This changes evil-visual-update-x-selection to use the PRIMARY
selection and also...

- Uses the wrapper `evil-set-selection` instead of the obsolete
  `x-select-text`
- Tests for `display-selections-p` instead of boundness of
  `x-select-text` or `ns-initialized`

### CODE REVIEW

**Comment by hpdeifel** [comment]
Tested with emacs-25 compiled with or without X and emacs-24 without X.

This would be a fix for bug #532

**Comment by hpdeifel** [comment]
Two things that I'm not quite sure about:

1. Should we respect `select-active-region`? I think every visual selection can be considered as *active* region in emacs lingo and I'm not sure if evil has a concept of transient-mark (as in: A mouse selection)
2. Is it even correct to call `evil-visual-update-x-selection` from an idle-timer? What if the user selects a frame on another display (e.g another X server) before the timer expires?

**Comment by olejorgenb** [comment]
1. My vote would be to treat every selection the same

File: evil-states.el [hunk 1]
```diff
@@ -350,17 +350,13 @@ otherwise exit Visual state."
 
 (defun evil-visual-update-x-selection (&optional buffer)
   "Update the X selection with the current visual region."
-  (let ((buf (or buffer (current-buffer))))
-    (when (buffer-live-p buf)
```
Comment (wasamasa) [inline evil-states.el]: Why did you remove the `buffer-live-p` check?

**Comment by wasamasa** [comment]
1. Evil intercepts setting the mark (be it with `C-SPC` or the mouse) and enables visual char state, so it doesn't even have the concept of a transient mark.  There's currently a bit of code dealing with `select-active-regions` in Evil for it to not interfere with other commands, so I wouldn't honor it.  In case we're violating Vim compatibility or want to support a Vim option where it would be useful to honor, we can still ponder this issue.
2. I'd hope so! If the user manages such a feat in less than 0.1s and runs into an error, they'll hopefully report a bug.  In other words, I don't care too much about unhatched eggs, proven bugs however...

File: evil-states.el (hunk 1 above)
Comment (hpdeifel) [reply evil-states.el]: Because I based the code on the version from 2012 that didn't have this and completely ignored this part of the diff. Thanks for noticing, I fixed it in a new commit (and added a test).

**Comment by wasamasa** [comment]
Tests keep running and the functionality works as expected.  The only annoyance left is that flycheck (curiously `make` doesn't) complains about `evil-set-selection` being possibly undefined, but I'll investigate into this.

**Comment by hpdeifel** [comment]
Thanks!
